core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -624,7 +624,19 @@ class WorkspaceService(protected val ctx: RawlsRequestContext,
       } yield ()
     }
 
-  def assertNoChildrenBlockingWorkspaceDeletion(workspace: Workspace): Future[Unit] = for {
+  def assertNoGoogleChildrenBlockingWorkspaceDeletion(workspace: Workspace): Future[Unit] = for {
+    _ <-
+      if (workspace.googleProjectId.value.isEmpty) {
+        Future.failed(
+          RawlsExceptionWithErrorReport(
+            ErrorReport(StatusCodes.InternalServerError,
+                        "Cannot call this method on a workspace with no googleProjectId"
```

Comment:
I like the assertion. Do you know if we log internal server errors? I'd like to put some info in the logs because this condition should never really be hit (we should have determined that it's an azure or mc workspace before this function is called).
Could you add a log statement at some appropriate level (error maybe?) to this and include the workspace name (ie `namespace/name`) + any other information you think useful?

Reply:
It will get logged. I added the workspaceId to the message, and that should be sufficient.

Reply:
Workspace id is a little painful as it's hard to query for; it's stored as binary and rawls' mysql version lacks the helpful utils to convert uuids.

Reply:
We very commonly log IDs, and I search for them quite often… albeit, not through a direct SQL query.

Reply:
In any case, I would not expect that we will ever actually hit this line of code. It's here mostly to ensure that our existing test coverage ensures this method isn't called for Azure workspaces (as the Sam stuff is mocked out, and thus doesn't blow up in the way that the call to real Sam does).